### DISCUSSION TITLE
Cannot set custom error for array field if validation rule is defined for that field.

### DISCUSSION THREAD

**Discussion by chetankondawle** [react-hook-form/react-hook-form#10474] [Q&A] [answered]
### Version Number

7.43..9

### Codesandbox/Expo snack

https://codesandbox.io/s/customvalidationissue-xv2eol

### Steps to reproduce

1. Go to [https://codesandbox.io/s/customvalidationissue-xv2eol](https://codesandbox.io/s/customvalidationissue-xv2eol)
2. Click on Add
3. Type chetan in First Name

Actual : I am performing check to see if any field value has duplicate or not for firstName  in array. Code works fine if I do not set any validation rule in TextfieldInput custom component. But when validation rule is added, custom error with type unique is not being set. 

Error object is empty but the condition gets satisied as seen in console.log

### Expected behaviour

Error object should be populated with error type as unique for given field, regardless of validation rule is set or not.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Supriya21412** [comment]
Did you find the solution yet?
Kind off facing the same issue

**Comment by leapful** [comment, marked as answer]
@chetankondawle @Supriya21412 

There are some issues on the codesandbox sample like setting "shouldUnregister: true" on `useFieldArray` hook and also `useWatch` on field `users` will not work correctly on change of field `users.${index}.firstName` will perform validation at field level, not on field array level.

Furthermore, built-in validation rules from react-hook-form will perform after `setError` and it will trigger clearing errors on the field following the convention of custom error on registered field.

A workaround for this case is to use `setTimeout` for `setError` call so it can be executed after built-in validation rules.

Working demo: https://codesandbox.io/s/customvalidationissue-forked-ztwlzf?file=/src/App.tsx

**Reply by chetankondawle** [reply]
Thank for your insight and suggestion and the solution. 

Do you think this could be a feature request? To have unqiue value as validation rules for usefieldArray
